From 78d71268f7b967f0eea7860e3d10d0e97a2360d0 Mon Sep 17 00:00:00 2001
From: Philip Walden <pwaldenlinux@comcast.net>
Date: Thu, 8 Mar 2012 16:10:32 -0800
Subject: [PATCH] Fixed ustream file names and updated ustream gateway server
 address

Changed the filename title extraction to use the meta title tag.
Used wireshark to identify the new ustream gateway server.
---
 lib/FlashVideo/Site/Ustream.pm |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/lib/FlashVideo/Site/Ustream.pm b/lib/FlashVideo/Site/Ustream.pm
index e58b617..2ec8ed3 100644
--- a/lib/FlashVideo/Site/Ustream.pm
+++ b/lib/FlashVideo/Site/Ustream.pm
@@ -18,7 +18,7 @@ GHJwaW4uMC4xODM2MDk4NTkzMTY0Njg5OAAHdmlkZW9JZAIABzIzNTU3MzYAB3BhZ2VVcmwCACZo
 dHRwOi8vd3d3LnVzdHJlYW0udHYvcmVjb3JkZWQvMjM1NTczNgAHYnJhbmRJZAIAATEAAAkK
 EOF
 
-  my($title) = $browser->content =~ /<h2[^>]*>([^<]+)/;
+  my $title = extract_info($browser)->{meta_title};
 
   # http://www.ustream.tv/recorded/\d+
   my($video_id) = $browser->uri =~ m{recorded/(\d+)};
@@ -31,7 +31,7 @@ EOF
   $browser->post(
     # This is hidden as gwUrl inside the second loaded SWF
     # (viewer.rsl.VER.swf), too much effort to extract properly.
-    "http://216.52.240.138/gateway.php",
+    "http://rgw.ustream.tv/gateway.php",
     Content_Type => "application/x-amf",
     Content => $data
   );
-- 
1.7.4.4

